### PR TITLE
Support symfony 7

### PR DESCRIPTION
**Pull request by Wiktor6**
AC
---
* support symfony 7
* drop support for symfony < 6
* remove deprecated src/LuigisBoxBundle/Service/ConfigProvider::getRequestHeaders

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -1,29 +1,20 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/6.1/phpunit.xsd"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.5/phpunit.xsd"
```
Comment (lukasz-falda) [inline phpunit.xml.dist]: use `vendor/phpunit/phpunit/phpunit.xsd` instead

File: src/LuigisBoxBundle/Exception/ApiErrorException.php [hunk 2]
```diff
@@ -8,20 +8,10 @@
...
 {
-    /**
-     * @var Request
-     */
-    private $request;
-
-    public function __construct(string $message, Request $request)
+    public function __construct(
+        string $message,
+        public readonly Request $request)
```
Comment (lukasz-falda) [inline src/LuigisBoxBundle/Exception/ApiErrorException.php]: format

File: src/LuigisBoxBundle/Exception/MalformedResponseException.php [hunk 3]
```diff
@@ -9,31 +9,12 @@
...
-     * @var Request
-     */
-    private $request;
-
-    public function __construct(string $message, ResponseInterface $response, Request $request, ?\Throwable $previous = null)
+    public function __construct(
+        string $message,
+        public readonly ResponseInterface $response,
+        public readonly Request $request,
+        ?\Throwable $previous = null)
```
Comment (lukasz-falda) [inline src/LuigisBoxBundle/Exception/MalformedResponseException.php]: format

File: src/LuigisBoxBundle/Service/ConfigProvider.php [hunk 4]
```diff
@@ -111,31 +110,18 @@ public function setHeader(string $name, string $value): void
...
-    }
-
     public function resetHeaders(): void
     {
         $this->headers = [];
     }
 
-    /**
-     * @deprecated use getAuthorizationHeaders method instead
-     */
```
Comment (lukasz-falda) [inline src/LuigisBoxBundle/Service/ConfigProvider.php:126]: write about it on release note

File: src/LuigisBoxBundle/ValueObject/ContentRemovalCollection.php [hunk 5]
```diff
@@ -6,30 +6,15 @@
...
-class ContentRemovalCollection implements ObjectsInterface, \Countable
+readonly class ContentRemovalCollection implements ObjectsInterface, \Countable
 {
-    /**
-     * @var ContentRemoval[]
-     */
-    private $objects;
-
-    public function __construct(array $objects)
+    public function __construct(public array $objects)
```
Comment (lukasz-falda) [inline src/LuigisBoxBundle/ValueObject/ContentRemovalCollection.php:14]: `ContentRemoval` `@param`

File: src/LuigisBoxBundle/ValueObject/SearchUrlBuilder.php [hunk 6]
```diff
@@ -160,10 +101,7 @@ public function setFilters(array $filters): self
         return $this;
     }
 
-    /**
-     * @param bool|int|string $value
-     */
-    public function addMustFilter(string $key, $value): self
+    public function addMustFilter(string $key, bool|int|string $value): self
     {
```
Comment (lukasz-falda) [inline src/LuigisBoxBundle/ValueObject/SearchUrlBuilder.php:105]: below conditions can be removed

File: tests/Acceptance/Service/RequestTest.php [hunk 7]
```diff
@@ -34,75 +37,176 @@ protected function setUp(): void
         $this->configProvider = ExampleConfiguration::provideDefaultConfig();
     }
 
-    /**
-     * @test
-     * @dataProvider \Answear\LuigisBoxBundle\Tests\DataProvider\RequestDataProvider::forContentUpdate()
-     */
+    #[Test]
+    #[DataProvider('forContentUpdate')]
```
Comment (lukasz-falda) [inline tests/Acceptance/Service/RequestTest.php]: why not from `RequestDataProvider`? It is impossible to load from external static method?

**Review by bmulawa — APPROVED** [review]

File: src/LuigisBoxBundle/ValueObject/SearchUrlBuilder.php (hunk 6 above)
Comment (Wiktor6) [reply src/LuigisBoxBundle/ValueObject/SearchUrlBuilder.php:105]: Then it will throw different exception, is it ok?

**Review by lukasz-falda — APPROVED** [review]

**Pull request merged**